testing/fds.go

```diff
@@ -0,0 +1,52 @@
+package testing
+
+import (
+	"net"
+	"os"
+)
+
+type Fds struct{}
```

Comment:
I think it's OK to merge testing.Fds and testing.Upgrader into a single struct. User code will always want to use them as a single interface, no?

Reply:
There are some people who [call `Upgrader.Fds.Listen()`](https://github.com/sagikazarmark/modern-go-application/blob/64673035cc3ceead255e1cd675f46a8febaeb53a/cmd/modern-go-application/main.go#L198), and this way will allow them to keep doing that if they want. (It means the interface they have to write will be more complicated, but ensures backwards compatibility.)